From b560444c66b6a52d6270f8509e2c26f55f29f167 Mon Sep 17 00:00:00 2001
From: Mingliang Liu <liuml07@apache.org>
Date: Thu, 16 Feb 2017 16:51:03 -0800
Subject: [PATCH 2265/2848] HADOOP-14040. Use shaded aws-sdk uber-JAR 1.11.86.
 Contributed by Steve Loughran and Sean Mackrory

This also removes previous shading done on the AWS SDK JARs in CDH.

(cherry picked from commit 658702efffdf52cf5ddf8e92f959f1157c95a348)

Conflicts:
	hadoop-project/pom.xml
	hadoop-tools/hadoop-aws/pom.xml

Change-Id: I1cf7631ba1b737de714dc16fe81e8ffc7fe6f2fa
---
 hadoop-project/pom.xml          |   14 ++-------
 hadoop-tools/hadoop-aws/pom.xml |   62 +--------------------------------------
 2 files changed, 3 insertions(+), 73 deletions(-)

diff --git a/hadoop-project/pom.xml b/hadoop-project/pom.xml
index b78777b..6f648c9 100644
--- a/hadoop-project/pom.xml
+++ b/hadoop-project/pom.xml
@@ -98,7 +98,7 @@
     <maven-surefire-plugin.version>${cdh.surefire.version}</maven-surefire-plugin.version>
     <maven-surefire-report-plugin.version>${maven-surefire-plugin.version}</maven-surefire-report-plugin.version>
     <maven-failsafe-plugin.version>${maven-surefire-plugin.version}</maven-failsafe-plugin.version>
-    <aws-java-sdk.version>1.10.6</aws-java-sdk.version>
+    <aws-java-sdk.version>1.11.86</aws-java-sdk.version>
     <surefire.fork.timeout>900</surefire.fork.timeout>
   </properties>
 
@@ -678,17 +678,7 @@
       </dependency>
       <dependency>
         <groupId>com.amazonaws</groupId>
-        <artifactId>aws-java-sdk-s3</artifactId>
-        <version>${aws-java-sdk.version}</version>
-      </dependency>
-      <dependency>
-        <groupId>com.amazonaws</groupId>
-        <artifactId>aws-java-sdk-dynamodb</artifactId>
-        <version>${aws-java-sdk.version}</version>
-      </dependency>
-      <dependency>
-        <groupId>com.amazonaws</groupId>
-        <artifactId>aws-java-sdk-sts</artifactId>
+        <artifactId>aws-java-sdk-bundle</artifactId>
         <version>${aws-java-sdk.version}</version>
       </dependency>
       <dependency>
diff --git a/hadoop-tools/hadoop-aws/pom.xml b/hadoop-tools/hadoop-aws/pom.xml
index 851d4ed..f8d32dd 100644
--- a/hadoop-tools/hadoop-aws/pom.xml
+++ b/hadoop-tools/hadoop-aws/pom.xml
@@ -351,50 +351,6 @@
       </plugin>
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
-        <artifactId>maven-shade-plugin</artifactId>
-        <executions>
-          <execution>
-            <phase>package</phase>
-            <goals>
-              <goal>shade</goal>
-            </goals>
-            <configuration>
-              <createDependencyReducedPom>false</createDependencyReducedPom>
-              <artifactSet>
-                <includes>
-                  <include>com.amazonaws:aws-java-sdk-sts:jar:*</include>
-                  <include>com.amazonaws:aws-java-sdk-s3:jar:*</include>
-                  <include>com.amazonaws:aws-java-sdk-dynamodb:jar:*</include>
-                  <include>com.amazonaws:aws-java-sdk-core:jar:*</include>
-                  <include>com.fasterxml.jackson.*:*</include>
-                  <include>joda-time:joda-time:jar:*</include>
-                  <include>org.apache.httpcomponents:*:jar:*</include>
-                </includes>
-              </artifactSet>
-              <relocations>
-                <relocation>
-                  <pattern>com.fasterxml.jackson</pattern>
-                  <shadedPattern>com.cloudera.com.fasterxml.jackson</shadedPattern>
-                </relocation>
-                <relocation>
-                  <pattern>com.amazonaws</pattern>
-                  <shadedPattern>com.cloudera.com.amazonaws</shadedPattern>
-                </relocation>
-                <relocation>
-                  <pattern>org.apache.http</pattern>
-                  <shadedPattern>com.cloudera.org.apache.http</shadedPattern>
-                </relocation>
-                <relocation>
-                  <pattern>org.joda</pattern>
-                  <shadedPattern>com.cloudera.org.joda</shadedPattern>
-                </relocation>
-              </relocations>
-            </configuration>
-          </execution>
-        </executions>
-      </plugin>
-      <plugin>
-        <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-surefire-plugin</artifactId>
         <configuration>
           <forkedProcessTimeoutInSeconds>3600</forkedProcessTimeoutInSeconds>
@@ -488,14 +444,8 @@
     </dependency>
     <dependency>
       <groupId>com.amazonaws</groupId>
-      <artifactId>aws-java-sdk-s3</artifactId>
+      <artifactId>aws-java-sdk-bundle</artifactId>
       <scope>compile</scope>
-      <!-- If you exclude joda time here, it will *not* be included in the
-	   uber/shaded jar.  It happened to work with older versions of
-	   aws-java-sdk probably because joda time wasn't actually used.
-	   Instead, artifacts that have dependency on this .jar should add
-	   exclusions for joda-time.
-      -->
     </dependency>
     <dependency>
       <groupId>com.fasterxml.jackson.core</groupId>
@@ -514,16 +464,6 @@
       <artifactId>joda-time</artifactId>
     </dependency>
     <dependency>
-      <groupId>com.amazonaws</groupId>
-      <artifactId>aws-java-sdk-dynamodb</artifactId>
-      <scope>compile</scope>
-    </dependency>
-    <dependency>
-      <groupId>com.amazonaws</groupId>
-      <artifactId>aws-java-sdk-sts</artifactId>
-      <scope>compile</scope>
-    </dependency>
-    <dependency>
       <groupId>junit</groupId>
       <artifactId>junit</artifactId>
       <scope>test</scope>
-- 
1.7.9.5

